ros2/ament.bzl

```diff
@@ -147,6 +147,18 @@ def _ros2_ament_setup_rule_impl(ctx):
         )
         outputs.append(dynamic_library)
 
+    for dep in ctx.attr.idl_deps:
+        package_name = dep[Ros2IdlPluginAspectInfo].plugins.to_list()[-1].package_name
+        for src in dep[Ros2InterfaceInfo].info.srcs:
+            src_file = ctx.actions.declare_file(
+                paths.join(prefix_path, "share", package_name, "msg", src.basename),
```

Comment:
I would also declare ament package xmls if not already defined (take a look above a bit).

Reply:
Another thing: `idl.srcs` has all kinds of IDL files (actions, msgs, srvs), so I'd propose you filter here `.msg` files here.